### DISCUSSION TITLE
Vendor class isochronous endpoint

### DISCUSSION THREAD

**Discussion by DarthAffe** [hathach/tinyusb#744] [Q&A] [answered]
Hi
I'm trying to use an Raspberry Pi Pico as a vendor class device with isochronous out transfer (using libusb on windows).
Since I couldn't find anything regarding isochronous transfer aside from the audio-class I tried to use that to adapt the vendor descriptor but I can't get it to work.

It's detected as a USB device in the device manager but it can't be enabled (the same code works fine using the default vendor descriptor).

Am I missing something or isn't this possible with tinyusb?

---

That's what I came up with as descriptor (it contains in and out endpoints even if I only need out since it didn't load at all without the in endpoint):
```
#define TUD_VENDOR_ISO_DESCRIPTOR(_itfnum, _stridx, _epin, _epout, _epsize, _interval) \
    9, TUSB_DESC_INTERFACE, _itfnum, 0, 2, TUSB_CLASS_VENDOR_SPECIFIC, 0x00, 0x00, _stridx, \
    7, TUSB_DESC_ENDPOINT, _epout, (TUSB_XFER_ISOCHRONOUS | TUSB_ISO_EP_ATT_NO_SYNC | TUSB_ISO_EP_ATT_DATA), U16_TO_U8S_LE(_epsize), _interval, \
    7, TUSB_DESC_ENDPOINT, _epin, (TUSB_XFER_ISOCHRONOUS | TUSB_ISO_EP_ATT_NO_SYNC | TUSB_ISO_EP_ATT_DATA), U16_TO_U8S_LE(_epsize), _interval
```

**Comment by hathach** [comment, marked as answer]
You need to write your own driver for your special need, here is how rpi team add their https://github.com/raspberrypi/pico-sdk/pull/197 . Also ISO transfer is not fully support as well.
